### PR TITLE
#159082381 Add average rating output on user rating output

### PR DESCRIPTION
**Pull request by abulojoshua1**
### What does this PR do?
- Returns average rating on an article
- Returns all the data of the rating that has just been created

### Description of Task to be completed?
- A user is able to rate an article.

### How should this be manually tested?
- Create a rating request in the format shown below.
```
{
    "Rating": {
		"rating": 4,
		"review": "The service was great"
    }
}
```
- Output should be shown as in the example below
```
{
    "Rating": {
        "rating": "4",
        "review": "The service was great",
        "article_id": 6,
        "user_id": 1,
        "message": "Article rated successfully.",
        "average_rating": 3.8620689655172415
    }
}
```
- Send it to the API endpoint `/api/articles/<int: article_id>/rating/`
- Note: review can contain one insight on the article but it is not a must to add it.

### Any background context you want to provide?
- I modified the renderer class to return rating
- Changed some lines of code in tests to suit all changes made
- Calculated average rating on an article
### What are the relevant pivotal tracker stories?
- [#159082381](https://www.pivotaltracker.com/story/show/159082381)

### CODE REVIEW

**Comment by geofrocker** [comment]
Rebase and resove the merge conflicts

**Comment by abulojoshua1** [comment]
Ok, I will resolve those merge conflicts ASAP. l

File: authors/apps/articles/renderers.py [hunk 1]
```diff
@@ -45,3 +45,25 @@ def render(self, data, media_type=None, renderer_context=None):
...
+
+    def render(self, data, media_type=None, renderer_context=None):
+        # If the view throws an error (such as the user can't be authenticated
+        # or something similar), `data` will contain an `errors` key. We want
+        # the default JSONRenderer to handle rendering errors, so we need to
+        # check for this case.
+        errors = data.get('errors', None)
+
+        if errors is not None:
+            pass
```
Comment (ssewilliam) [inline authors/apps/articles/renderers.py]: Is this a typo for the condition?

**Review by ssewilliam — APPROVED** [review]
@abulojoshua1 
Looks ok even without reusability

- Just fix the typo

File: authors/apps/articles/renderers.py (hunk 1 above)
Comment (abulojoshua1) [reply authors/apps/articles/renderers.py]: I got this. I will fix that ASAP!

**Comment by geofrocker** [comment]
Rename this PR appropriately

**Pull request merged**